enos/modules/test_cluster_health/scripts/servers.sh

```diff
@@ -9,36 +9,61 @@ error_exit() {
     exit 1
 }
 
+MAX_WAIT_TIME=40
+POLL_INTERVAL=2
+
+elapsed_time=0
+
 # Quality: nomad_agent_info: A GET call to /v1/agent/members returns the correct number of running servers and they are all alive
 
-servers=$(nomad server members -json )
-running_servers=$(echo $servers | jq '[.[] | select(.Status == "alive")]')
-servers_length=$(echo "$running_servers" | jq 'length' )
+while true; do  
+    servers=$(nomad operator autopilot health -json)
+    servers_healthy=$(echo "$servers" | jq -r '[.Servers[] | select(.Healthy == true) | .ID] | length')
 
-if [ -z "$servers_length" ];  then
-    error_exit "No servers found"
-fi
+    if [ "$servers_healthy" -eq 0 ]; then
+        error_exit "No servers found."
+    fi
 
-if [ "$servers_length" -ne "$SERVER_COUNT" ]; then
-    error_exit "Unexpected number of servers are alive: $servers_length\n$(echo $servers | jq '.[] | select(.Status != "alive") | .Name')"
-fi
+    if [ "$servers_healthy" -eq "$SERVER_COUNT" ]; then
+        break
+    fi
 
-# Quality: nomad_agent_info_self: A GET call to /v1/agent/self against every server returns the same last_log_index for all of them"
+    if [ "$elapsed_time" -ge "$MAX_WAIT_TIME" ]; then
+        error_exit "Unexpected number of healthy servers: $servers_healthy after $elapsed_time seconds."
+    fi
 
-last_index=""
+    echo "Servers found: $servers_healthy, expected: $SERVER_COUNT. Waiting for $elapsed_time seconds. Retrying in $POLL_INTERVAL seconds..."
+    sleep "$POLL_INTERVAL"
+    elapsed_time=$((elapsed_time + POLL_INTERVAL))
+done
+# Quality: nomad_agent_info_self: A GET call to /v1/agent/self against every server returns the same last_log_index for all of them"
```

Comment:
We're actually checking that every server is are equally caught up to the log index checked at the start, which only checks that we're _roughly_ caught up (which is good!). I like the approach we're taking here, but maybe we should make the comment more clear.

Reply:
Is the updated one better?